registry_auth/auth.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
```

Comment:
IIRC some of the implementation details here were based on hard to find documentation.  Can we include the a link to the relevant documentation as a comment?